tests/suites/test_suite_des.function

```diff
@@ -14,6 +14,119 @@ void des_check_weak( data_t * key, int ret )
+    TEST_INVALID_PARAM_RET( MBEDTLS_ERR_DES_BAD_INPUT_DATA,
+                            mbedtls_des3_set2key_enc( NULL, key ) );
+    TEST_INVALID_PARAM_RET( MBEDTLS_ERR_DES_BAD_INPUT_DATA,
+                            mbedtls_des3_set2key_enc( &ctx_3des, NULL ) );
+    TEST_INVALID_PARAM_RET( MBEDTLS_ERR_DES_BAD_INPUT_DATA,
+                            mbedtls_des3_set2key_dec( NULL, key ) );
+    TEST_INVALID_PARAM_RET( MBEDTLS_ERR_DES_BAD_INPUT_DATA,
+                            mbedtls_des3_set2key_dec( &ctx_3des, NULL ) );
+    TEST_INVALID_PARAM_RET( MBEDTLS_ERR_DES_BAD_INPUT_DATA,
+                            mbedtls_des3_set3key_enc( NULL, key ) );
+    TEST_INVALID_PARAM_RET( MBEDTLS_ERR_DES_BAD_INPUT_DATA,
+                            mbedtls_des3_set3key_enc( &ctx_3des, NULL ) );
+    TEST_INVALID_PARAM_RET( MBEDTLS_ERR_DES_BAD_INPUT_DATA,
+                            mbedtls_des3_set3key_dec( NULL, key ) );
+    TEST_INVALID_PARAM_RET( MBEDTLS_ERR_DES_BAD_INPUT_DATA,
+                            mbedtls_des3_set3key_dec( &ctx_3des, NULL ) );
+
+    TEST_INVALID_PARAM_RET( MBEDTLS_ERR_DES_BAD_INPUT_DATA,
+                            mbedtls_des_crypt_ecb( NULL, input, output ) );
+    TEST_INVALID_PARAM_RET( MBEDTLS_ERR_DES_BAD_INPUT_DATA,
+                            mbedtls_des_crypt_ecb( &ctx_des, NULL, output ) );
+    TEST_INVALID_PARAM_RET( MBEDTLS_ERR_DES_BAD_INPUT_DATA,
+                            mbedtls_des_crypt_ecb( &ctx_des, input, NULL ) );
+
+#if defined(MBEDTLS_CIPHER_MODE_CBC)
+    TEST_INVALID_PARAM_RET( MBEDTLS_ERR_DES_BAD_INPUT_DATA,
+                            mbedtls_des_crypt_cbc( NULL,
+                                                   MBEDTLS_DES_ENCRYPT,
+                                                   sizeof( input ), iv, input,
+                                                   output )  );
+    TEST_INVALID_PARAM_RET( MBEDTLS_ERR_DES_BAD_INPUT_DATA,
+                            mbedtls_des_crypt_cbc( &ctx_des,
+                                                   MBEDTLS_DES_ENCRYPT,
+                                                   sizeof( input ), NULL, input,
+                                                   output )  );
+    TEST_INVALID_PARAM_RET( MBEDTLS_ERR_DES_BAD_INPUT_DATA,
+                            mbedtls_des_crypt_cbc( &ctx_des,
+                                                   MBEDTLS_DES_ENCRYPT,
+                                                   sizeof( input ), iv, NULL,
+                                                   output )  );
+    TEST_INVALID_PARAM_RET( MBEDTLS_ERR_DES_BAD_INPUT_DATA,
+                            mbedtls_des_crypt_cbc( &ctx_des,
+                                                   MBEDTLS_DES_ENCRYPT,
+                                                   sizeof( input ), iv, input,
+                                                   NULL )  );
+#endif /* MBEDTLS_CIPHER_MODE_CBC */
+
+    TEST_INVALID_PARAM_RET( MBEDTLS_ERR_DES_BAD_INPUT_DATA,
```

Comment:
Shouldn't this test be guarded with the define `!defined(MBEDTLS_DES3_CRYPT_ECB_ALT)` as the function itself?

Reply:
No: While we're omitting the API function declarations if `MBEDTLS_XXX_ALT` is set, we require users to declare and implement the same API in this case, and so the tests still apply.